globalnoc-tsds-datasource/src/datasource.js

```diff
@@ -399,8 +404,21 @@ export class GenericDatasource {
     }
 
     buildQueryParameters(options, t) {
+	
+      // function returns template variables and its selected value i.e. {'metric':'average', 'example': 'another_metric'}
+      function getVariableDetails(){
+    	let varDetails = {};
+    	t.templateSrv.variables.forEach(function(item){
+      	  if(item.type === "custom"){
```

Comment:
We don't need to be checking for the variable type, since we don't care how it was populated. Pull in every template variable here should be fine.